.github/workflows/testing.yml

```diff
@@ -81,25 +88,17 @@ jobs:
         && source /opt/conda/etc/profile.d/conda.sh \
         && source /opt/conda/etc/profile.d/mamba.sh \
         && mamba activate mss-${{ inputs.branch_name }}-env \
-        && xvfb-run pytest -v --durations=20 --reverse --cov=mslib tests \
-        || (for i in {1..5} \
-          ; do xvfb-run pytest tests -v --durations=0 --reverse --last-failed --lfnf=none \
-          && break \
-        ; done)
+        && xvfb-run pytest -v --durations=20 --reverse --cov=mslib tests
 
     - name: Run tests in parallel
       if: ${{ success() && inputs.xdist == 'yes' }}
-      timeout-minutes: 25
+      timeout-minutes: 5
       run: |
         cd $GITHUB_WORKSPACE \
         && source /opt/conda/etc/profile.d/conda.sh \
         && source /opt/conda/etc/profile.d/mamba.sh \
         && mamba activate mss-${{ inputs.branch_name }}-env \
-        && xvfb-run pytest -vv -n 6 --dist loadfile --max-worker-restart 4 tests \
-        || (for i in {1..5} \
-          ; do xvfb-run pytest -vv -n 6 --dist loadfile --max-worker-restart 0 tests --last-failed --lfnf=none \
-          && break \
-        ; done)
+        && xvfb-run pytest -vv -n 6 --dist loadfile --max-worker-restart 4 tests
```

Comment:
the 4 is for four times restarting, do we keep this?

Reply:
I kept it for now because I didn't want to change the xdist related flags. I have a follow up PR to explore if those can be improved.